File: Data Structures and Algorithms/4. Recursion/Ankon Sir Problems/Common Problems/_2power.cpp
#include <iostream>
using namespace std;

int power(int x, int y) {
    if(y == 1)
        return x;

    return power(x, y - 1) * x;
}

int main () {
    int x, y;
    cin >> x >> y;

    cout << x << " raised to the power " << y << " is " << power(x, y) << '\n';
    return 0;
}

File: Data Structures and Algorithms/4. Recursion/Ankon Sir Problems/Common Problems/_2power.java
import java.util.*;

class _2power {
    public static int power(int x, int y) {
        if(y == 1) return x;
        return power(x, y - 1) * x;
    }

    public static void main(String[] args) {
        int x, y;
        Scanner sc = new Scanner(System.in);
        x = sc.nextInt();
        y = sc.nextInt();

        System.out.printf("%d raised to the power %d is %d \n", x, y, power(x, y));
    }
}
